code/modules/admin/view_variables/view_variables.dm

```diff
@@ -78,11 +82,18 @@
 			var/name = dropdownoptions[i]
 			var/link = dropdownoptions[name]
 			dropdownoptions[i] = "<option value[link? "='[link]'":""]>[name]</option>"
+	else if(isappearance)
+		dropdownoptions = vv_get_dropdown_appearance(thing)
 	else
 		dropdownoptions = thing.vv_get_dropdown()
 
 	var/list/names = list()
-	if(!islist)
+	if(isappearance)
+		var/static/list/virtual_appearance_vars
+		if(!virtual_appearance_vars)
+			virtual_appearance_vars = build_virtual_appearance_vars()
```

Comment:
```suggestion
		var/static/list/virtual_appearance_vars = null
		if(isnull(virtual_appearance_vars))
			virtual_appearance_vars = build_virtual_appearance_vars()
```

this is a lot more explicit whenever people need to look at this code in the future (and `isnull()` is very slightly faster than `!` anyways, it's not the point but it's just a nice thing)

Reply:
actually I changed it into `var/static/list/virtual_appearance_vars = build_virtual_appearance_vars()`
I don't get myself why I didn't do that at first lol